src/node_buffer.cc

```diff
@@ -541,17 +541,17 @@ void WriteDoubleBE(const FunctionCallbackInfo<Value>& args) {
 }
 
 
-void ByteLength(const FunctionCallbackInfo<Value> &args) {
+void ByteLengthUTF8(const FunctionCallbackInfo<Value> &args) {
   Environment* env = Environment::GetCurrent(args);
 
   if (!args[0]->IsString())
     return env->ThrowTypeError("Argument must be a string");
 
-  Local<String> s = args[0]->ToString(env->isolate());
-  enum encoding e = ParseEncoding(env->isolate(), args[1], UTF8);
+  // Fast case: avoid StringBytes on UTF8 string. Jump to v8.
+  Local<String> str = args[0]->ToString(env->isolate());
```

Comment:
Yea the string check and env can be dropped, this is always called with a string from js.